src/torcontrol.cpp

```diff
@@ -631,10 +634,14 @@ void TorController::disconnected_cb(TorControlConnection& _conn)
     if (!reconnect)
         return;
 
-    LogDebug(BCLog::TOR, "Not connected to Tor control port %s, trying to reconnect\n", m_tor_control_center);
-
     // Single-shot timer for reconnect. Use exponential backoff.
-    struct timeval time = MillisToTimeval(int64_t(reconnect_timeout * 1000.0));
+    int64_t timeout = std::min(reconnect_timeout, RECONNECT_TIMEOUT_MAX);
+    int64_t timeout_ms = int64_t(timeout * 1000.0);
+    struct timeval time = MillisToTimeval(timeout_ms);
+
+    LogDebug(BCLog::TOR, "Not connected to Tor control port %s, retrying in %.2f seconds\n",
+             m_tor_control_center, timeout);
+
     if (reconnect_ev)
         event_add(reconnect_ev, &time);
     reconnect_timeout *= RECONNECT_TIMEOUT_EXP;
```

Comment:
imo it would be better to do the capping here:
```c++
reconnect_timeout = std::min(reconnect_timeout * RECONNECT_TIMEOUT_EXP, RECONNECT_TIMEOUT_MAX);
```

Reply:
Thank you, updated.